modules/ui/fields/wikipedia.js

```diff
@@ -307,15 +297,25 @@ export function uiFieldWikipedia(field, context) {
     }
   }
 
-  function updateEncodedWikiUrl(tagLang, tagArticleTitle, anchor) {
-    const underscoredUrlEncodedTitle = encodeURIComponent(tagArticleTitle.replace(/ /g, '_'));
-    const urlAnchored = toUrlAnchor(anchor);
-    _wikiURL = `${scheme}${tagLang}.${domain}/wiki/${underscoredUrlEncodedTitle}${urlAnchored}`;
-  }
+  wiki.encodePath = (tagArticleTitle, anchor) => {
+    const underscoredTitle = tagArticleTitle.replace(/ /g, '_');
+    return `${encodeURIComponent(underscoredTitle)}${anchor ? '#' + encodeURIComponent(wiki.anchorFragment(anchor)): ''}`;
+  };
 
-  function toUrlAnchor(anchor) {
-    return anchor ? '#' + anchor : '';
-  }
+  wiki.anchorFragment = (anchor) => {
+    if (!anchor) {
+      return '';
+    }
+
+    try {
+      // Best-effort `anchorencode:` implementation
+      anchor = anchor.replace(/ /g, '_').replace(/%/g, '.');
+    } catch (e) {
```

Comment:
I refactored it and yes I believe it could go away, so it is now gone. Please note that _try-catch_ `anchorencode` was in there before this fix. It got extracted into a separate function without  affecting the flow.

I dug further into it, and couldn't find any reason* for replacing `%` with `.`

I concluded that it was a Best-effort at trying to deal with URI encoded looking characters that happen to be contained in sections then made anchors, without resorting to `encodeURIComponent()`, here also for reasons I could not explain. Also, could not think of any possible string content that would make `replace()` throw an error, it may throw error on `null`, `undefined` value, and invalid regex though. But then it would throw an error within the catch block.

Anyhow I have added further unit test cases, which hopefully clarifies intents of those functions now. If the test cases aren't correct, happy to iterate again.


*or was it perhaps an attempt to defend against [Double Encoding Attacks](https://owasp.org/www-community/Double_Encoding) ? If that could be the case perhaps it should be put back where it was, which was the rational of keeping it in there initially, but in any case, it does not seem like a solid way to prevent URL tempering  attacks so I settled with eliminating.

Reply:
It looks like my recollection of MediaWiki’s behavior may be somewhat outdated. If a section is titled something like “[#time](https://www.mediawiki.org/wiki/Help:Extension:ParserFunctions##time)”, there will be two named anchors, `#time` and `.23time`. Some other reserved characters behave similar. It used to be the case that the page would contain only `.23time`, but now that it contains `#time` as well, there’s no need for anything special, other than replacing spaces with underscores.